### PR TITLE
ADR on args referencing x and y

### PR DESCRIPTION
**Pull request by SimonHeybrock**

### CODE REVIEW

File: docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+Particularly in scipp's plotting functionality there can be confusion from different interpretation of ``x`` or ``y``:
+
+- Dimensions ``'x'`` or ``'y'`` of the data being handled (plotted).
+- ``x`` and ``y`` axes of a (matplotlib) figure, e.g., as referenced by the ``xlim`` and ``ylim`` args matplotlib provides.
+- The meaning of above ``x`` and ``y`` and the configured limits after the "transpose" button in the plot was activated.
+
+Furthermore it should be considered that scipp supports dimensions in arbitrary order.
+That is, data depending on ``dim0`` and ``dim1`` may be stored as ``[dim0, dim1]`` or as ``[dim1, dim0]``.
+Creating a plot from this with hypothetical ``xlim`` or ``xmin`` arguments to ``plot()`` would thus yield different and undesirable results depending on an in principle irrelevant detail of the data.
```
Comment (nvaytet) [inline docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst:19]:
```suggestion
Creating a plot from this with hypothetical ``xlim`` or ``xmin`` arguments to ``plot()`` would thus yield different and unpredictable results depending on an, in principle, irrelevant detail of the data.
```

File: docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+The ambiguity can be avoided using scipp's ubiquitous approach of requiring explicit dimension labels.
+One example for this is the ``scale`` argument of ``plot()``, which is a dictionary mapping from dimension labels to the desired scale (linear or logarithmic).
+The same approach should be adopted for all other arguments to ``plot()``.
```
Comment (nvaytet) [inline docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst:23]: You could also mention the solution that uses slicing to provide the limits?

File: docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Furthermore it should be considered that scipp supports dimensions in arbitrary order.
+That is, data depending on ``dim0`` and ``dim1`` may be stored as ``[dim0, dim1]`` or as ``[dim1, dim0]``.
+Creating a plot from this with hypothetical ``xlim`` or ``xmin`` arguments to ``plot()`` would thus yield different and unpredictable results depending on an in principle irrelevant detail of the data.
+
+The ambiguity can be avoided using scipp's ubiquitous approach of requiring explicit dimension labels.
+One example for this is the ``scale`` argument of ``plot()``, which is a dictionary mapping from dimension labels to the desired scale (linear or logarithmic).
+The same approach should be adopted for all other arguments to ``plot()``.
+
+Note that in the case of defining limits and alternative approach recommended to users is the use of slicing.
```
Comment (nvaytet) [inline docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst:25]:
```suggestion
Note that in the case of defining limits, an alternative approach recommended to users is the use of slicing.
```

**Review by nvaytet — APPROVED** [review]

**Comment by jl-wynen** [comment]
What is the goal here? Is this meant to be a guide on how to add more functionality to plots? It looks like what is mentioned here is not really implemented yet.

**Comment by SimonHeybrock** [comment]
> What is the goal here? Is this meant to be a guide on how to add more functionality to plots? It looks like what is mentioned here is not really implemented yet.

The goals of ADRs is:
- To avoid having the same discussion 5 times.
- Having something to refer to for future additions, e.g., when adding new arguments to functions.
- Having something to refer users to if new feature requests are made, i.e., "won't fix", see ADR XYZ.
- Finally, if something existing is in conflict with an ADR it should be changed.
- Probably more, just google?

Of course ADR are not necessarily final, with good reasons old ADRs can be overruled. They simply represent an architectural decision at a certain time.

**Comment by nvaytet** [comment]
To be a little more specific to this ADR, we have had several people requesting plotting options such as xmin , xmax, ymin, ymax, or logx logy.
However, if you have coordinates that are named 'x' and 'y', and you can have coordinate 'x' along the vertical axis, then it is not clear if the xmin limit should apply to the 'x' coordinate or to the horizontal axis.
Also it is not clear what happens if you hit the transpose axes button, does the xmin follow the dimension or does it stay with the plot axis?
So writing this we can in the future refer users with similar requests to this page citing our reasons for not supporting these arguments.

File: docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+Particularly in scipp's plotting functionality there can be confusion from different interpretation of ``x`` or ``y``:
+
+- Dimensions ``'x'`` or ``'y'`` of the data being handled (plotted).
+- ``x`` and ``y`` axes of a (matplotlib) figure, e.g., as referenced by the ``xlim`` and ``ylim`` args matplotlib provides.
+- The meaning of above ``x`` and ``y`` and the configured limits after the "transpose" button in the plot was activated.
+
+Furthermore it should be considered that scipp supports dimensions in arbitrary order.
+That is, data depending on ``dim0`` and ``dim1`` may be stored as ``[dim0, dim1]`` or as ``[dim1, dim0]``.
+Creating a plot from this with hypothetical ``xlim`` or ``xmin`` arguments to ``plot()`` would thus yield different and unpredictable results depending on an in principle irrelevant detail of the data.
```
Comment (OwenArnold) [inline docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst:19]:
```suggestion
Creating a plot from this with hypothetical ``xlim`` or ``xmin`` arguments to ``plot()`` would thus yield different and unpredictable results depending on an, in principle, irrelevant detail of the data.
```

**Review by OwenArnold — APPROVED** [review]
Looks good other than minor punctation suggestion.

File: docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst (hunk 4 above)
Comment (SimonHeybrock) [reply docs/developer/adr/0007-do-not-support-args-referencing-x-or-y.rst]: 👍

**Pull request merged**